components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -24,6 +25,8 @@ BraveWalletService::BraveWalletService(
       brave_wallet::Network::kMainnet, url_loader_factory);
   keyring_controller_ =
       std::make_unique<brave_wallet::KeyringController>(prefs);
+  tx_controller_ = std::make_unique<brave_wallet::EthTxController>(
+      base::AsWeakPtr(this), prefs);
```

Comment:
I don't understand the point of this. It's impossible for `EthTxController` to outlive `BraveWalletService` because `tx_controller_` is a unique_ptr owned by `BraveWalletService`. Also we never check `wallet_service_` for null in `EthTxController` so the `WeakPtr` doesn't help anything. I think we should also avoid the circular dependency here since `BraveWalletService` is only used to get other controllers inside `EthTxController`

Reply:
issue opened in  brave/brave-browser#16823